File: Graduation_Java/app/src/main/java/com/example/graduation_java/m_UI/MyViewHolder.java
package com.example.graduation_java.m_UI;

import android.view.View;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.example.graduation_java.R;

public class MyViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

    TextView nameTxt, addressTxt, heightTxt;
    ItemClickListener itemClickListener;

    public MyViewHolder(View itemView) {
        super(itemView);

        nameTxt = (TextView)itemView.findViewById(R.id.tv_title);
        addressTxt = (TextView)itemView.findViewById(R.id.tv_addr);
        heightTxt = (TextView)itemView.findViewById(R.id.tv_hgt);

        itemView.setOnClickListener(this);
    }

    public void setItemClickListener(ItemClickListener itemClickListener){
        this.itemClickListener = itemClickListener;
    }

    @Override
    public void onClick(View view) {
        this.itemClickListener.onItemClick(this.getLayoutPosition());
    }
}

File: OpenApi_Kotlin/app/src/main/java/com/example/openapi_kotlin/m_UI/MyViewHolder.kt
package com.example.openapi_kotlin.m_UI

import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.openapi_kotlin.R

class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
    var nameTxt: TextView
    var addressTxt: TextView
    var heightTxt: TextView
    private var itemClickListener: ItemClickListener? = null
    fun setItemClickListener(itemClickListener: ItemClickListener?) {
        this.itemClickListener = itemClickListener
    }

    override fun onClick(view: View) {
        itemClickListener!!.onItemClick(this.layoutPosition)
    }

    init {
        nameTxt = itemView.findViewById<View>(R.id.tv_title) as TextView
        addressTxt = itemView.findViewById<View>(R.id.tv_addr) as TextView
        heightTxt = itemView.findViewById<View>(R.id.tv_hgt) as TextView
        itemView.setOnClickListener(this)
    }
}